### PR TITLE
[elf] Implement --export-dynamic-symbol[-list]. 

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
See descriptions in commit messages.

Closes: #532

### CODE REVIEW

File: elf/linker-script.cc [hunk 1]
```diff
@@ -348,6 +348,14 @@ void parse_version_script(Context<E> &ctx, std::string path) {
     SyntaxError(ctx, tok[0]) << "trailing garbage token";
 }
 
+template <typename E>
+void parse_dynamic_glob(Context <E> &ctx, std::string_view arg, bool is_cpp) {
+  if (arg == "*")
+    ctx.default_version_from_glob = VER_NDX_GLOBAL;
```
Comment (rui314) [inline elf/linker-script.cc:354]: I believe linker scripts are parsed after `parse_nonpositional_args`, so isn't it too late to assign a value to this variable?

Comment (ishitatsuyuki) [reply elf/linker-script.cc:354]: I changed their parsing order in https://github.com/rui314/mold/pull/588/commits/0525da2c0385a5498e8b66c2df121737a417a863.

Comment (ishitatsuyuki) [reply elf/linker-script.cc:354]: (Or did you actually meant linker script not version script? IIRC linker scripts are not implemented yet?)

Comment (rui314) [reply elf/linker-script.cc:354]: Linker script is supported, though we support only a limited set of directive such as `INPUT`, `OUTPUT_FORMAT` or `VERSION`. And with `VERSION`, you can embed a version script into a linker script.

Comment (ishitatsuyuki) [reply elf/linker-script.cc:354]: OK, I see, will try to get this fixed.

Comment (ishitatsuyuki) [reply elf/linker-script.cc]: I've reverted to parsing these as positional args in the latest revision.

**Pull request merged**